src/signature.c

```diff
@@ -361,13 +387,13 @@ static bool validate_certificate(void)
 		X509_STORE_CTX_free(verify_ctx);
 	}
 
-	return false;
+	return verify_ctx->error;
 }
 
 int verify_callback(int ok, X509_STORE_CTX *stor)
 {
 	if (!ok) {
-		fprintf(stderr, "Certificate verification error: %s\n",
+		printf("Certificate verification error: %s\n",
```

Comment:
We can keep this for consistency for now then. A revamp of the printing/error output seems needed overall.